From 5a29c3d4ab353fb4923edee497663d120fc66456 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Sat, 5 Feb 2022 18:24:42 +0100
Subject: [PATCH] Reset if medium not found or loading kernel/dtb failed

After multiple reboots it may happen that cboot can't find
the kernel and dtb partitions on the SD-CARD.

Let's reset in these cases.

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 bootloader/partner/t18x/cboot/platform/tegra_shared/debug.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/bootloader/partner/t18x/cboot/platform/tegra_shared/debug.c b/bootloader/partner/t18x/cboot/platform/tegra_shared/debug.c
index a1cc9ca..b174854 100644
--- a/bootloader/partner/t18x/cboot/platform/tegra_shared/debug.c
+++ b/bootloader/partner/t18x/cboot/platform/tegra_shared/debug.c
@@ -15,6 +15,7 @@
 #include <platform_c.h>
 #include <printf.h>
 #include <tegrabl_timer.h>
+#include <tegrabl_exit.h>
 #include <tegrabl_debug.h>
 
 #if defined(CONFIG_DEBUG_TIMESTAMP)
@@ -91,7 +92,9 @@ int platform_dgetc(char *c, bool wait)
 
 void platform_halt(void)
 {
-	dprintf(ALWAYS, "HALT: spinning forever...\n");
+	dprintf(ALWAYS, "Will reset in 10 seconds...\n");
+	tegrabl_mdelay(10 * 1000);
+	tegrabl_reset();
 	for(;;);
 }
 
-- 
2.17.1

